src/pages/core/using-core-in-anchor.md

```diff
@@ -0,0 +1,146 @@
+
+Start by borrowing the data inside the asset/collection account using the `try_borrow_data()` function. Then, create the asset/collection struct from those bytes like the example:
+
+{% dialect-switcher title="Deserialize an Account" %}
+
+{% dialect title="Asset" id="asset" %}
+
+```rust
+let data = ctx.accounts.asset.try_borrow_data()?;
+let base_asset: BaseAssetV1 = BaseAssetV1::from_bytes(&data.as_ref())?;
+```
+
+{% /dialect %}
+
+{% dialect title="Collection" id="collection" %}
+
+```rust
+let data = ctx.accounts.collectino.try_borrow_data()?;
+let base_collection: BaseCollectionV1 = BaseCollectionV1::from_bytes(&data.as_ref())?;
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+### Anchor Struct Method
+
+To deserialize assets in the account struct, enable the `anchor` feature in the mpl-core crate by modifying the cargo.toml like this:
+
+```toml
+[dependencies]
+...
+mpl-core = { version = "x.x.x", features = [ "anchor" ] }
+```
+
+By activating this feature, you'll be able to deserialize both the `Asset` and `Collection` accounts directly in the Anchor account struct:
+
+{% dialect-switcher title="Deserialize an Asset" %}
+
+{% dialect title="Asset" id="asset" %}
+
+```rust
+#[derive(Accounts)]
+pub struct ExampleAccountStruct<'info> {
+    ...
+    pub asset: Account<'info, BaseAssetV1>,
+}
+```
+
+{% /dialect %}
+
+{% dialect title="Collection" id="collection" %}
+
+```rust
+#[derive(Accounts)]
+pub struct ExampleAccountStruct<'info> {
+    ...
+    pub collection: Account<'info, BaseCollectionV1>,
+}
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+### Deserializing Plugins
+
+To access individual plugins within an Asset or Collection account, use the `fetch_plugin()` function. This function will either return the plugin data or a `null` response without throwing an hard error, allowing you to check if a plugin exists without having to access its data.
+
+The `fetch_plugin()` function is used for both Assets and Collections accounts and can handle every plugin type by specifying the appropriate typing. If you want to access the data inside a plugin, use the middle value returned by this function.
+
+{% dialect-switcher title="Deserialize an Asset" %}
+
+{% dialect title="Asset" id="asset" %}
+
+```rust
+let (_, attribute_list, _) = fetch_plugin::<BaseAssetV1, Attributes>(&ctx.accounts.asset.to_account_info(), mpl_core::types::PluginType::Attributes)?;
+```
+
+{% /dialect %}
+
+{% dialect title="Collection" id="collection" %}
+
+```rust
+let (_, attribute_list, _) = fetch_plugin::<BaseCollectionV1, Attributes>(&ctx.accounts.asset.to_account_info(), mpl_core::types::PluginType::Attributes)?;
+```
+
+{% /dialect %}
+
+{% /dialect-switcher %}
+
+**Note**: The `fetch_plugin()` function is only used for non-external plugins. To read external plugins, use the `fetch_external_plugin()` function, which operates in the same way as `fetch_plugin()`.
+
+## The CPI Instruction Builders
+
```

Comment:
````suggestion
### Instruction CPI Builders

- [AddCollectionExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.AddCollectionExternalPluginAdapterV1CpiBuilder.html)
- [AddCollectionPluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.AddCollectionPluginV1CpiBuilder.html)
- [AddExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.AddExternalPluginAdapterV1CpiBuilder.html)
- [AddPluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.AddPluginV1CpiBuilder.html)
- [ApproveCollectionPluginAuthorityV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.ApproveCollectionPluginAuthorityV1CpiBuilder.html)
- [ApprovePluginAuthorityV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.ApprovePluginAuthorityV1CpiBuilder.html)
- [BurnCollectionV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.BurnCollectionV1CpiBuilder.html)
- [BurnV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.BurnV1CpiBuilder.html)
- [CollectCpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.CollectCpiBuilder.html)
- [CreateCollectionV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.CreateCollectionV1CpiBuilder.html)
- [CreateCollectionV2CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.CreateCollectionV2CpiBuilder.html)
- [CreateV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.CreateV1CpiBuilder.html)
- [CreateV2CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.CreateV2CpiBuilder.html)
- [RemoveCollectionExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RemoveCollectionExternalPluginAdapterV1CpiBuilder.html)
- [RemoveCollectionPluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RemoveCollectionPluginV1CpiBuilder.html)
- [RemoveExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RemoveExternalPluginAdapterV1CpiBuilder.html)
- [RemovePluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RemovePluginV1CpiBuilder.html)
- [RevokeCollectionPluginAuthorityV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RevokeCollectionPluginAuthorityV1CpiBuilder.html)
- [RevokePluginAuthorityV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.RevokePluginAuthorityV1CpiBuilder.html)
- [TransferV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.TransferV1CpiBuilder.html)
- [UpdateCollectionExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdateCollectionExternalPluginAdapterV1CpiBuilder.html)
- [UpdateCollectionPluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdateCollectionPluginV1CpiBuilder.html)
- [UpdateCollectionV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdateCollectionV1CpiBuilder.html)
- [UpdateExternalPluginAdapterV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdateExternalPluginAdapterV1CpiBuilder.html)
- [UpdatePluginV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdatePluginV1CpiBuilder.html)
- [UpdateV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.UpdateV1CpiBuilder.html)
- [WriteCollectionExternalPluginAdapterDataV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.WriteCollectionExternalPluginAdapterDataV1CpiBuilder.html)
- [WriteExternalPluginAdapterDataV1CpiBuilder](https://docs.rs/mpl-core/0.7.2/mpl_core/instructions/struct.WriteExternalPluginAdapterDataV1CpiBuilder.html)
````

Reply:
This may look too large, have a see.

Reply:
In my opinion it's too large, but i left it there as comment so if you feel like it's good enough i can just uncomment it out